File: app/src/main/java/com/github/android/lvrn/lvrnproject/view/adapter/datapostset/DataPostSetAdapter.java
package com.github.android.lvrn.lvrnproject.view.adapter.datapostset;

import android.support.v7.widget.RecyclerView.Adapter;
import android.support.v7.widget.RecyclerView.ViewHolder;

import com.github.android.lvrn.lvrnproject.persistent.entity.ProfileDependedEntity;

import java.util.List;

/**
 * @author Vadim Boitsov <vadimboitsov1@gmail.com>
 */

public abstract class DataPostSetAdapter<T1 extends ProfileDependedEntity, T2 extends ViewHolder> extends Adapter<T2> {

    public abstract void setData(List<T1> data);

}

File: app/src/main/java/com/github/android/lvrn/lvrnproject/view/adapter/datapostset/DataPostSetAdapter.kt
package com.github.android.lvrn.lvrnproject.view.adapter.datapostset

import com.github.valhallalabs.laverna.persistent.entity.base.ProfileDependedEntity

/**
 * @author Vadim Boitsov <vadimboitsov1@gmail.com>
 */

interface DataPostSetAdapter<T1 : ProfileDependedEntity> {

    fun setData(data: List<T1>)
}